docs/installation.rst

```diff
@@ -2,11 +2,180 @@
 Installation
 ============
 
-At the command line::
+Install
+=======
 
-    $ easy_install puente
+Activate your virtual environment, then do:
 
-Or, if you have virtualenvwrapper installed::
+.. code-block:: bash
 
-    $ mkvirtualenv puente
-    $ pip install puente
+  $ pip install puente
+
+
+Configure
+=========
+
+Puente configuration goes in the ``PUENTE`` setting in your Django settings
+file. Here's a minimal example::
+
+   PUENTE = {
+       'ROOT': BASE_DIR,
+       'DOMAIN_METHODS': {
+           'django': [
+               ('jinja2/*.html', 'puente.extract.extract_jinja2'),
+               ('*.py', 'puente.extract.extract_python')
+           ]
+       }
+   }
+
+
+This sets up string extraction for ``jinja2/*.html`` files using the Jinja2
+extractor, ``*.py`` files using the Python extractor and puts all those strings
+in ``django.po(t)`` files.
+
+
+.. py:data:: ROOT
+
+   :type: String
+   :default: None
+   :required: Yes
+
+
+   This is the absolute path to the root directory which has ``locale/`` in it.
+
+   For example::
+
+       /home/willkg/
+          - fjord/         <-- ROOT
+            - .git/
+            - locale/
+            - fjord/
+              - code!!!
+            - manage.py
+
+
+.. py:data:: DOMAIN_METHODS
+
+   :type: Dict of string to list of (string, string) tuples
+   :default: None
+   :required: Yes
+
+
+   Dict of domain name to list of (file matcher, extractor) tuples.
+
+   A domain name here is the name that's used to name the ``.pot`` and ``.po``
+   files. For example, if the domain was "django", then the resulting files
+   would be ``django.pot`` and ``django.po``.
+
+   The file matcher uses ``*`` and ``**`` glob patterns.
+
+   The extractor is the Babel extractor function used to extract the strings
+   from files of that type. Puente helpfully centralizes these in the
+   ``puente.extract`` module:
+
+   * ``puente.extract.extract_jinja2``: Extracts strings from Jinja2 templates.
+   * ``puente.extract.extract_python``: Extracts strings from Python code
+     files.
+
+   For example:
+
+   .. code-block:: python
+
+      PUENTE = {
+          'DOMAIN_METHODS': {
+              'django': [
+                  ('jinja2/*.html', 'puente.extract.extract_jinja2'),
+                  ('*.py', 'puente.extract.extract_python')
+              ]
+          }
+      }
+
+
+.. py:data:: KEYWORDS
+
+   :type: List of strings
+   :default: Common gettext indicators
+   :required: No
+
+
+   The list of keywords for functions that are gettext-related. This defaults to
+   the list Babel has plus ``_lazy``. If you find it doesn't include all the
```

Comment:
I  think it would be good to not reference Babel here. I don't know what "the list Babel has" means, at least. Maybe give a full list? Or say something like "This includes common gettext functions, such as `_`, `_lazy`, and `gettext`."?